test-data/unit/daemon.test

```diff
@@ -522,3 +522,20 @@ class A:
     x: int
 class B:
     x: int
+
+[case testUnusedTypeIgnorePreservedAfterUpdate]
+-- Regression test for https://github.com/python/mypy/issues/9655
+$ dmypy start -- --warn-unused-ignores --no-error-summary
+Daemon started
+$ dmypy check -- bar.py
+bar.py:2: error: Unused "type: ignore" comment
+== Return code: 1
+$ dmypy check -- bar.py
+bar.py:2: error: Unused "type: ignore" comment
+== Return code: 1
+
+[file foo/__init__.py]
+[file foo/empty.py]
+[file bar.py]
+from foo.empty import *
+a = 1  # type: ignore
```

Comment:
**Question:** I have discovered that I can reproduce this with any unused import, so long as it's from a nested module.

For example, this also fails:

```
[file bar.py]
from http.client import HTTPConnection
a = 1  # type: ignore
```

This requires fewer files to be created, but presumes the existence of a module in the standard library, which may confuse the intent of the test.

At the risk of bikeshedding, is there a preferable approach in this codebase?

Reply:
Most tests in mypy's test suite should ideally not make any assumptions about which modules exist in the standard library (with a few exceptions such as `typing`, `typing_extensions`, etc), since typeshed's stubs for the standard library are constantly being improved and updated.

There are some exceptions to this rule — e.g. the tests in this file are only guaranteed to work if mypy uses all of typeshed's stdlib stubs: https://github.com/python/mypy/blob/master/test-data/unit/pythoneval.test

But generally speaking, we prefer to use that approach only if it's impossible to reproduce a bug properly without assuming certain stdlib features.

Reply:
That makes good sense to me. I'll leave the test as it is then. Thank you :)

Reply:
No worries!